src/braket/circuits/circuit.py

```diff
@@ -762,10 +763,16 @@ def measure(self, target_qubits: QubitSetInput | None = None) -> Circuit:
 
         if target_qubits:
             # Check if the target_qubits are already measured
-            if self._measure_targets and all(
-                target in self._measure_targets for target in target_qubits
+            if (
+                self._measure_targets
+                and any(target in self._measure_targets for target in target_qubits)
+                or len(target_qubits) != len(set(target_qubits))
             ):
-                intersection = set(target_qubits) & set(self._measure_targets)
+                intersection = (
+                    set(target_qubits) & set(self._measure_targets)
+                    if self._measure_targets
+                    else [qubit for qubit, count in Counter(target_qubits).items() if count > 1]
```

Comment:
I'd actually put these two cases in separate branches with different error messages, since one is for qubits that have already been measured, and another is for repeated qubits in the same measurement.

Reply:
Sound good, broke this into two different checks. One for qubits that are already measured and one if the qubit targets are repeated in the same measurement.